### PR TITLE
feat: add seed-dev-agent script for local dev setup (LPP-2.2)

### PR DESCRIPTION
**Pull request by dodizzle**
## Summary
- Adds `scripts/seed-dev-agent.ts` — an idempotent seed script that upserts a dev agent (`id: dev-agent`) into the admin Prisma DB with its env vars, plugin, and default tools
- Reads `CLAUDE_CODE_OAUTH_TOKEN` (required) and `GH_TOKEN` (optional) from `state/dev-agent.env`; exits non-zero with a clear actionable message if the file or token is missing
- Adds `state/dev-agent.env` to `.gitignore` to prevent credentials from being committed

## Acceptance Criteria

| # | Criterion | Status | Evidence |
|---|-----------|--------|----------|
| 1 | Idempotent: running twice produces the same DB state | MET | All writes use Prisma `upsert`; dedicated idempotency test verifies second run makes identical calls |
| 2 | GET /agents/dev-agent/config returns seeded env vars, plugin, and tools | MET | Agent upserted with `id="dev-agent"`; env, plugin, tools seeded as related records the endpoint reads |
| 3 | Missing state/dev-agent.env exits non-zero with clear actionable message | MET | `exit(1)` with multi-line instructions; unit test covers this path |
| 4 | Unit test with injected Prisma double covering all upserts + error exits | MET | 5 tests using injected doubles cover Agent+AgentEnv+AgentPlugin+AgentTool upserts, missing-file, missing-token, and idempotency |
| 5 | state/dev-agent.env added to .gitignore | MET | `.gitignore` diff adds the entry |

## Test Plan
- [x] `bun test scripts/seed-dev-agent.unit.test.ts` — 5/5 pass
- [x] `bunx biome lint .` — no issues
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/seed-dev-agent.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  exit: (code: number, message: string) => never;
+}
+
+// ─── Env file parser ──────────────────────────────────────────────────────────
+
+/**
+ * Parse a dotenv-style file into a key/value map.
+ * Ignores blank lines and lines starting with #.
+ */
+export function parseEnvFile(content: string): Record<string, string> {
```
Comment (dmcaulay) [inline scripts/seed-dev-agent.ts:95]: `parseEnvFile` is exported but has no dedicated unit tests. Values containing `=` (e.g. `KEY=abc=def`), unmatched quotes, and `\r\n` line endings aren't covered. The function is tested indirectly through `seedDevAgent`, but a brief `describe("parseEnvFile")` block would lock in these edge cases cheaply since the function is already exported.

**Review by dmcaulay — APPROVED** [review]
APPROVE — clean implementation. All 5 ACs met, CI green. Two minor suggestions (parseEnvFile edge-case tests, AC#2 integration coverage) — neither blocks merge.

**Review by dodizzle — COMMENTED** [review]
APPROVE — Clean, well-tested implementation. All 5 acceptance criteria met, CI green. Correct Prisma compound keys, no-leak `$disconnect()` in `finally`, DI architecture throughout with no module mocks. No blocking issues found. Human approval already in place from @dmcaulay. (Posted as COMMENT per self-review policy.)

**Review by dodizzle — COMMENTED** [review]
APPROVE — Suggestion addressed. 14 dedicated `parseEnvFile` unit tests added covering `=` in values, `\r\n` line endings, quote-stripping edge cases, blank lines, comments, missing `=`, whitespace, and empty input. The documented unmatched-quote behavior matches the implementation exactly. No new issues. CI green. Clear to merge.

*(Posted as COMMENT per self-review policy — author == reviewer.)*

**Review by dmcaulay — APPROVED** [review]
APPROVE — Suggestion addressed. 14 dedicated `parseEnvFile` tests added covering `=` in values, `\r\n` line endings, quote-stripping, unmatched quotes, blank lines, comments, missing `=`, whitespace, and empty input. CI green. Clear to merge.
